### PR TITLE
bpo-41576: document BaseException in favor of bare except

### PR DESCRIPTION
**Pull request by graingert**
https://bugs.python.org/issue41576

### CODE REVIEW

File: Doc/tutorial/errors.rst [hunk 1]
```diff
@@ -162,9 +160,12 @@ the exception (allowing a caller to handle the exception as well)::
        print("OS error: {0}".format(err))
    except ValueError:
        print("Could not convert data to an integer.")
-   except:
-       print("Unexpected error:", sys.exc_info()[0])
+   except BaseException as err:
+       print("Unexpected error:", err)
        raise
+       
+Alternatively a the last except clause may omit the exception name(s), however the exception
```
Comment (graingert) [inline Doc/tutorial/errors.rst]:
```suggestion
Alternatively the last except clause may omit the exception name(s), however the exception
```

**Review by terryjreedy — APPROVED** [review]
To me, the change improves this section.

File: Doc/tutorial/errors.rst [hunk 2]
```diff
@@ -489,4 +490,3 @@ After the statement is executed, the file *f* is always closed, even if a
 problem was encountered while processing the lines. Objects which, like files,
 provide predefined clean-up actions will indicate this in their documentation.
```
Comment (graingert) [inline Doc/tutorial/errors.rst]:
```suggestion


```

File: Doc/tutorial/errors.rst [hunk 3]
```diff
@@ -162,10 +160,13 @@ the exception (allowing a caller to handle the exception as well)::
...
    except ValueError:
        print("Could not convert data to an integer.")
-   except:
-       print("Unexpected error:", sys.exc_info()[0])
+   except BaseException as err:
+       print("Unexpected error:", err)
        raise
 
+Alternatively the last except clause may omit the exception name(s), however the exception
+value must then be retrieved from ``sys.exc_info()[0]``.
```
Comment (graingert) [inline Doc/tutorial/errors.rst]:
```suggestion
value must then be retrieved from ``sys.exc_info()[1]``.
```

File: Doc/tutorial/errors.rst [hunk 4]
```diff
@@ -162,10 +160,13 @@ the exception (allowing a caller to handle the exception as well)::
        print("OS error: {0}".format(err))
    except ValueError:
        print("Could not convert data to an integer.")
-   except:
-       print("Unexpected error:", sys.exc_info()[0])
+   except BaseException as err:
+       print("Unexpected error:", err)
```
Comment (graingert) [inline Doc/tutorial/errors.rst]:
```suggestion
       print(f"Unexpected {err=}, {type(err)=}")
```

File: Doc/tutorial/errors.rst [hunk 5]
```diff
@@ -488,5 +489,3 @@ used in a way that ensures they are always cleaned up promptly and correctly. ::
 After the statement is executed, the file *f* is always closed, even if a
 problem was encountered while processing the lines. Objects which, like files,
 provide predefined clean-up actions will indicate this in their documentation.
```
Comment (graingert) [inline Doc/tutorial/errors.rst:498]: My editor keeps trimming this buggy whitespace, so we should probably remove it anyway

**Comment by graingert** [comment]
@terryjreedy what's the next step here?

**Comment by terryjreedy** [comment]
I am looking at this and will request or make some changes.  I need to look at the context and think of how to answer Serhiy's objection on the issue.

**Comment by graingert** [comment]
@terryjreedy I've now fixed the conflicts in this PR

**Comment by terryjreedy** [comment]
@ambv Thanks for finishing this.

**Pull request merged**